examples/flutter_gallery/lib/demo/text_field_demo.dart

```diff
@@ -69,6 +75,30 @@ class TextFieldDemoState extends State<TextFieldDemo> {
     return null;
   }
 
+  Future<bool> _warnUserAboutInvalidData() {
+    final FormState form = _formKey.currentState;
+    if (!_formWasEdited || form.validate())
+      return new Future<bool>.value(true);
+
+    return showDialog(
```

Comment:
Yes, it returns a Future whose value is the bool passed to the Navgator.pop() that dismisses the Alert dialog.